src/game-logic/reducers/upgradeTool.js

```diff
@@ -0,0 +1,24 @@
+import upgrades from '../../data/upgrades'
+import { TOOL_UPGRADED_NOTIFICATION } from '../../templates'
+
+import { showNotification } from './showNotification'
+import { makeRecipe } from './makeRecipe'
+
+/**
+ * @param {farmhand.state} state
+ * @param {farmhand.upgrade} upgrade
+ */
+export const upgradeTool = (state, upgrade) => {
```

Comment:
🎏 (oops this one was probably one I should have added!)

Reply:
Nobody's perfect! 😄

https://github.com/jeremyckahn/farmhand/commit/bd4a4469350942dfbdd713f9f520ded38a1ffed5